core/src/test/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerImplTest.java

```diff
@@ -0,0 +1,677 @@
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+public class AndroidMediaPlayerImplTest {
+
+    private static final boolean IS_BEATING = true;
+    private static final boolean IS_NOT_BEATING = false;
+
+    private static final long TWO_MINUTES_IN_MILLIS = 120000;
+    private static final long DELAY_MILLIS = 500;
+
+    private static final int WIDTH = 100;
+    private static final int HEIGHT = 200;
+    private static final int ANY_ROTATION_DEGREES = 0;
+    private static final int ANY_PIXEL_WIDTH_HEIGHT = 1;
+    private static final int TEN_SECONDS = 10;
+    private static final int ONE_SECOND_IN_MILLIS = 1000;
+
+    private static final boolean IS_PLAYING = true;
+    private static final Uri URI = Mockito.mock(Uri.class);
+    private static final Timeout ANY_TIMEOUT = Timeout.fromSeconds(TEN_SECONDS);
+    private static final Player.LoadTimeoutCallback ANY_LOAD_TIMEOUT_CALLBACK = new Player.LoadTimeoutCallback() {
+        @Override
+        public void onLoadTimeout() {
+
+        }
+    };
+    private static final PlayerAudioTrack PLAYER_AUDIO_TRACK = new PlayerAudioTrack(0, 0, "id", "english", ".mp4", 1, 120);
+    private static final List<PlayerAudioTrack> AUDIO_TRACKS = Collections.singletonList(PLAYER_AUDIO_TRACK);
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private AndroidMediaPlayerFacade mediaPlayer;
+    @Mock
+    private PlayerListenersHolder listenersHolder;
+    @Mock
+    private MediaPlayerForwarder forwarder;
+    @Mock
+    private LoadTimeout loadTimeout;
+    @Mock
+    private Heart heart;
+    @Mock
+    private Handler handler;
+    @Mock
+    private CheckBufferHeartbeatCallback bufferHeartbeatCallback;
+    @Mock
+    private BuggyVideoDriverPreventer buggyVideoDriverPreventer;
+    @Mock
+    private PreparedListeners preparedListeners;
+    @Mock
+    private BufferStateListeners bufferStateListeners;
+    @Mock
+    private ErrorListeners errorListeners;
+    @Mock
+    private CompletionListeners completionListeners;
+    @Mock
+    private VideoSizeChangedListeners videoSizeChangedListeners;
+    @Mock
+    private InfoListeners infoListeners;
+    @Mock
+    private StateChangedListeners stateChangedListeners;
+    @Mock
+    private Player.PreReleaseListener playerReleaseListener;
+    @Mock
+    private MediaPlayer.OnPreparedListener onPreparedListener;
+    @Mock
+    private MediaPlayer.OnCompletionListener onCompletionListener;
+    @Mock
+    private MediaPlayer.OnErrorListener onErrorListener;
+    @Mock
+    private MediaPlayer.OnVideoSizeChangedListener onSizeChangedListener;
+
+    @Mock
+    private CheckBufferHeartbeatCallback.BufferListener bufferListener;
+    private AndroidMediaPlayerImpl player;
+
+    @Before
+    public void setUp() {
+        Log.setShowLogs(false);
+        given(listenersHolder.getPreparedListeners()).willReturn(preparedListeners);
```

Comment:
Can we not use a real `listenersHolder` here instead? Would make this bit cleaner